### PR TITLE
Create a page to view the list of users

### PR DESCRIPTION
**Pull request by AndriiChepeliuk**
--- Add ---

- added page with list of users;
- added result page "403" if user are not allowed to some page
- added menuItems for IdentityRole "Logist"

--- Change ---

- services/authentication.js - go to page after authentication depending on the IdentityRole
- the "Add car" button was removed for "Logist"

### CODE REVIEW

File: src/services/users.js [hunk 1]
```diff
@@ -77,3 +77,15 @@ export async function getFullUserName() {
...
+    return await usersService
+        .getAllIdentityUsers()
+        .then(
+            (response) => {
+                return response.data;
+            }
+        )
+}
+
+
```
Comment (Grasmus) [inline src/services/users.js]: Remove empty line

File: src/services/users.js [hunk 2]
```diff
@@ -77,3 +77,15 @@ export async function getFullUserName() {
...
 }
+
+export async function getAllIdentityUsers() {
+    return await usersService
+        .getAllIdentityUsers()
+        .then(
+            (response) => {
+                return response.data;
+            }
+        )
```
Comment (Grasmus) [inline src/services/users.js:94]: What will happen if error occurs?

File: src/privateRoute.js [hunk 3]
```diff
@@ -9,7 +10,14 @@ const PrivateRoute = props => {
...
     if (isAuthUser && !allowedRoles.includes(role)) {
-        return <Route component={() => <Redirect to="/login" />} />;
+        return (
+            <Result
+                status="403"
+                title="403"
+                subTitle="You are not allowed to see this page."
+            />
+        );
+        //return <Route component={() => <Redirect to="/login" />} />;
```
Comment (Grasmus) [inline src/privateRoute.js]: What is it?

File: src/components/navigation/menu.jsx [hunk 4]
```diff
@@ -1,13 +1,31 @@
...
+import jwt from 'jwt-decode';
+import tokenService from "../../services/tokens";
 
 function Menu({ isMenuOpen, data }) {
     let history = useHistory();
+    let accessToken = tokenService.getLocalAccessToken();
+    let decodedAccessToken = jwt(accessToken);
+    let menuItems = menuUserItems;
+
+    switch (decodedAccessToken.role) {
```
Comment (Grasmus) [inline src/components/navigation/menu.jsx]: I think it is better here to use dictionary with key Role and value MenuItems

File: src/api/users.js [hunk 5]
```diff
@@ -13,4 +13,8 @@ export default class usersService {
   static getFullUserName() {
     return instance.get(USERS_URLS.FULL_NAME);
   }
+
+  static getAllIdentityUsers() {
```
Comment (marynakernychna) [inline src/api/users.js]: Why don't you call it "getAllUsers"?

File: src/components/navigation/menu.jsx [hunk 6]
```diff
@@ -1,13 +1,31 @@
...
 import user_icon from "../../assets/images/user.png";
 import { Link } from "react-router-dom";
 import { Button } from 'antd';
+import { userRoles } from '../../constants/userRoles';
+import jwt from 'jwt-decode';
+import tokenService from "../../services/tokens";
 
 function Menu({ isMenuOpen, data }) {
     let history = useHistory();
+    let accessToken = tokenService.getLocalAccessToken();
```
Comment (marynakernychna) [inline src/components/navigation/menu.jsx]: We should not work with tokens in components

File: src/components/navigation/menu.jsx (hunk 4 above)
Comment (marynakernychna) [inline src/components/navigation/menu.jsx]: You can get the role of a user from Redux, it is already recorded there

File: src/components/navigation/menuItems.js [hunk 7]
```diff
@@ -1,8 +1,9 @@
 import changeRoleMenuItem_icon from "../../assets/images/change.png";
 import userCarsMenuItem_icon_icon from "../../assets/images/car.png";
 import userOffersMenuItem_icon from "../../assets/images/location.png";
+import users_icon from "../../assets/images/users_icon.png"
```
Comment (marynakernychna) [inline src/components/navigation/menuItems.js]: Please change from .png to .svg (files not created by you will be modified later)

File: src/components/navigation/menuItems.js [hunk 8]
```diff
@@ -1,8 +1,9 @@
...
 
-export const menuItems = [
+export const menuUserItems = [
```
Comment (marynakernychna) [inline src/components/navigation/menuItems.js]: Why don't you call it "userMenuItems"?

File: src/components/navigation/menuItems.js [hunk 9]
```diff
@@ -19,3 +20,11 @@ export const menuItems = [
         itemLink: "/my-offers"
     }
 ];
+
+export const menuLogistItems = [
```
Comment (marynakernychna) [inline src/components/navigation/menuItems.js]: Why don't you call it "logistMenuItems"?

File: src/components/users/index.jsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+function IdentityUserListPage() {
+
+    const [users, setUsers] = useState([]);
+
+    useEffect(async () => {
+        setUsers(await getAllIdentityUsers());
+    }, []);
+
+    return (
+        <div className="allIdentityUsersBody">
```
Comment (marynakernychna) [inline src/components/users/index.jsx]: Why don't you call it "usersPageBody"?

File: src/components/users/index.jsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+                        <User info={user} />
+                    )}
+                </div>
+                :
+                <Result
+                    status="404"
+                    title="There are no registered users yet!"
+                />
+            }
+
```
Comment (marynakernychna) [inline src/components/users/index.jsx]: Remove the extra empty ine

File: src/components/users/index.jsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useState } from "react";
+import { getAllIdentityUsers } from "../../services/users";
+import { Result } from "antd";
+import User from "./user";
+import Header from "../navigation/header";
+
+function IdentityUserListPage() {
```
Comment (marynakernychna) [inline src/components/users/index.jsx]: Why don't you call it "ManageUsersPage"?

File: src/components/users/user/index.jsx [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+                <div className="rightSide">
+                    <Button
+                        className="detailsButton"
+                        type="primary">
+                        Details
+                    </Button>
+                </div>
+            </div>
+
```
Comment (marynakernychna) [inline src/components/users/user/index.jsx]: Remove the extra empty line

File: src/constants/api/urls.js [hunk 14]
```diff
@@ -40,5 +40,6 @@ export const TRIPS_URL = {
 export const USERS_URLS = {
     PROFILE_INFO: USER_URL + "/user-info",
     EDIT: USER_URL + "/edit-info",
-    FULL_NAME: USER_URL + "/user-full-name"
+    FULL_NAME: USER_URL + "/user-full-name",
+    GET_ALL_IDENTITY_USERS: USER_URL
```
Comment (marynakernychna) [inline src/constants/api/urls.js]: Why don't you call it "GET_ALL_USERS"?

File: src/reducers/authReducer.js [hunk 15]
```diff
@@ -34,6 +34,21 @@ const authReducer = (state = intialState, action) => {
                 }
             }
 
+            if (decodedAccessToken.role === userRoles.LOGIST) {
```
Comment (marynakernychna) [inline src/reducers/authReducer.js]: It is necessary to make from 2 if — one in which condition will be checked whether the role is in an array of userRoles. And in the body — tokens are set and change the role & the isAuthUser from the state

File: src/services/authentication.js [hunk 16]
```diff
@@ -55,8 +56,19 @@ export function login(values, history) {
         .then(
             (response) => {
                 store.dispatch(setAccess(response.data));
-
-                history.push("/main");
+                let accessToken = tokenService.getLocalAccessToken();
+                let decodedAccessToken = jwt(accessToken);
+                switch (decodedAccessToken.role) {
```
Comment (marynakernychna) [inline src/services/authentication.js]: You can get the role of a user from Redux, it is already recorded there

File: src/services/authentication.js [hunk 17]
```diff
@@ -55,8 +56,19 @@ export function login(values, history) {
...
+                    case userRoles.USER:
+                        history.push("/main");
+                        break;
+                    case userRoles.LOGIST:
+                        history.push("/users");
+                        break;
+                    default:
+                        history.push("/login");
```
Comment (marynakernychna) [inline src/services/authentication.js]: I think you should display an error message

File: src/services/users.js [hunk 18]
```diff
@@ -77,3 +77,15 @@ export async function getFullUserName() {
             );
         });
 }
+
+export async function getAllIdentityUsers() {
```
Comment (marynakernychna) [inline src/services/users.js]: Why don't you call it "getAllUsers"?

**Review by marynakernychna — CHANGES_REQUESTED** [review]
Also, the Logist button should not display the "Add car" button, so that when you click on it, an error window will be displayed

**Review by marynakernychna — APPROVED** [review]

**Review by Grasmus — APPROVED** [review]

**Pull request merged**